src/main/java/gregtech/api/recipes/ModHandler.java

```diff
@@ -337,6 +345,83 @@ public static Object finalizeIngredient(Object ingredient) {
         return ingredient;
     }
 
+    public static ItemMaterialInfo getRecyclingIngredients(Object... recipe) {
+        Map<Character, Integer> inputCountMap = new HashMap<>();
+        Map<Material, Long> materialStacksExploded = new HashMap<>();
+
+        int itr = 0;
+        while (recipe[itr] instanceof String) {
+            String s = (String) recipe[itr];
+            for (char c : s.toCharArray()) {
+                if (Character.isLowerCase(c)) continue; // skip tools
+                int count = inputCountMap.getOrDefault(c, 0);
+                inputCountMap.put(c, count + 1);
+            }
+            itr++;
+        }
+
+        char lastChar = ' ';
+        for (int i = itr; i < recipe.length; i++) {
+            Object ingredient = recipe[i];
+
+            // Track the current working ingredient symbol
+            if (ingredient instanceof Character) {
+                lastChar = (char) ingredient;
+                continue;
+            }
+
+            // Should never happen if recipe is formatted correctly
+            // In the case that it isn't, this error should be handled
+            // by an earlier method call parsing the recipe.
+            if (lastChar == ' ') return null;
```

Comment:
What about recipe objects that have a `' '` in them? Something like 

```
'WC'
'S '
```
To create a 2x2 recipe with the bottom right space being empty. Or something in the shape of the armor chestplate, which has the top middle slot as a `' '`.

Reply:
This is in the section parsing the `key, value` pairs for the recipe nputs, not the pattern. Empty spaces in the pattern will pass through fine